src/main/java/com/castletroymedical/billing/InvoiceBuilder.java

```diff
@@ -0,0 +1,71 @@
+package com.castletroymedical.billing;
+
+import com.castletroymedical.billing.invoice.Consultation;
+import com.castletroymedical.billing.invoice.InsuredInvoice;
+import com.castletroymedical.billing.invoice.Invoice;
+import com.castletroymedical.billing.invoice.MedicalCardInvoice;
+import com.castletroymedical.billing.invoice.Monitoring;
+import com.castletroymedical.billing.invoice.PrivateInvoice;
+import com.castletroymedical.billing.invoice.PrivateRoom;
+import com.castletroymedical.billing.invoice.Procedure;
+import com.castletroymedical.billing.invoice.WardBed;
+
+public class InvoiceBuilder {
+
+    private Invoice invoice;
+    
+    public InvoiceBuilder(String type){ 
+        this.invoice = invoiceFactory(type);
+    }
+
+    public Invoice invoiceFactory(String type){
+        if(type == null || type.isEmpty())
+            return null;
+
+        Invoice out;
+        type = type.toLowerCase();
+        switch(type) {
+            case "insured": out = new InsuredInvoice(); break;
+            case "medical card": out = new MedicalCardInvoice(); break;
+            default: out = new PrivateInvoice(); break;
```

Comment:
Does this mean that any other string provided to the factory will result in a privateInvoice being created?

Reply:
Yes, technically. If we're giving the front side only a choice of 3 which is "insured", "medical card" and "private," there's no other kind of input but those, was my logic for it. I can easily change it out to be private tho, no bother.

Reply:
I moreso meant have a case for private and the default case be null. Changing it to private would be a good idea too.